docs/feature_midi.md

```diff
@@ -0,0 +1,299 @@
+
+# MIDI
+
+## Enabling MIDI
+
+Add the following to the keyboard's rules.mk
+
+```c
+MIDI_ENABLE = yes
+```
+
+## Using Basic MIDI
+
+Enabling Basic MIDI only
+
+Add the following to your keyboard's config.h
+
+```c
+#define MIDI_BASIC
+```
+
+With basic MIDI you will only be able to send `NOTE_ON` and `NOTE_OFF` Commands to the internal MIDI device, meaning that keycodes like `MI_OCTU` and `MI_OCTD` won't work.
+
+## Advanced MIDI
+
+Enabling Advanced MIDI
+
+Add the following to your keyboard's config.h
+
+```c
+#define MIDI_ADVANCED
+```
+
+With advanced MIDI you can do things like octave shifts, channel changes, velocity changes, and modulation.
+
+### Sending MIDI Control Codes (MIDI CC)
+
+If you're aiming to emulate the features of something like a Launchpad or other MIDI controller you'll need to access the internal MIDI device directly.
+
+It's best to follow a refernce because when sending MIDI CC you don't get the advantages of a keycode and you will need to impliment a `process_record_user` and the custom keycodes for that. It's the same process as implmenting custom key behavior or macros.
```

Comment:
```suggestion
It's best to follow a reference because when sending MIDI CC you don't get the advantages of a keycode and you will need to implement a `process_record_user` and the custom keycodes for that. It's the same process as implementing custom key behavior or macros.
```
What do you mean by "implement a `process_record_user`" ?

Reply:
To implement custom midi keycodes, you need to use process record user. I'll clarify the wording.